src/renderer/gl_engine/tvgGlShaderSrc.h

```diff
@@ -30,5 +30,14 @@ extern const char* LINEAR_GRADIENT_FRAG_SHADER;
 extern const char* RADIAL_GRADIENT_FRAG_SHADER;
 extern const char* IMAGE_VERT_SHADER;
 extern const char* IMAGE_FRAG_SHADER;
+extern const char* MASK_VERT_SHADER;
+extern const char* MASK_ALPHA_FRAG_SHADER;
+extern const char* MASK_INV_ALPHA_FRAG_SHADER;
+extern const char* MASK_LUMA_FRAG_SHADER;
+extern const char* MASK_INV_LUMA_FRAG_SHADER;
+extern const char* MASK_ADD_FRAG_SHADER;
+extern const char* MASK_SUB_FRAG_SHADER;
+extern const char* MASK_INTERSECT_FRAG_SHADER;
+extern const char* MASK_DIFF_FRAG_SHADER;
 
 #endif /* _TVG_GL_SHADERSRC_H_ */
```

Comment:
Should we use 8 different shaders, or maybe use a single shader with switch inside of it?

Reply:
Complex if-else branch in fragment shader may negatively impact performance. 
And also need additional uniform variable to control these branch.

Consider the performance and for clean shader code, I choose split these branches into sperate shaders

Reply:
Maybe we can use them in the shader code when shaders are quite simlar.
 #if  ~   #elif  ~  #endif